### PR TITLE
Suggest associated const for incorrect use of let in traits

### LINKED ISSUE #101797: Suggest associated const for incorrect use of `let` in traits
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=3ee5f1570a9f674943f2f756d464877b)

```rust
trait Trait {
    let x: i32;
}

fn main() {

}
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: non-item in item list
 --> src/main.rs:2:5
  |
1 | trait Trait {
  |             - item list starts here
2 |     let x: i32;
  |     ^^^ non-item starts here
3 | }
  | - item list ends here

error: could not compile `playground` due to previous error
```

Ideally the output should suggest changing the `let` to a `const` for associated const. Thanks.

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #101797

### CODE REVIEW

File: compiler/rustc_parse/src/parser/item.rs [hunk 1]
```diff
@@ -698,11 +698,22 @@ impl<'a> Parser<'a> {
...
+
                     let mut err = self.struct_span_err(non_item_span, "non-item in item list");
-                    err.span_label(open_brace_span, "item list starts here")
-                        .span_label(non_item_span, "non-item starts here")
-                        .span_label(self.prev_token.span, "item list ends here");
+                    self.consume_block(Delimiter::Brace, ConsumeClosingDelim::Yes);
+                    if is_let {
+                        err.span_suggestion(
+                            non_item_span,
+                            "considering use `const` instead of `let` for associated const",
```
Comment (Rageking8) [inline compiler/rustc_parse/src/parser/item.rs:708]: "consider using" sounds better than "considering use" IMO.

File: src/test/ui/parser/suggest-assoc-const.stderr [hunk 2]
```diff
@@ -0,0 +1,8 @@
+error: non-item in item list
+  --> $DIR/suggest-assoc-const.rs:4:5
+   |
+LL |     let _X: i32;
+   |     ^^^ help: considering use `const` instead of `let` for associated const: `const`
```
Comment (Rageking8) [inline src/test/ui/parser/suggest-assoc-const.stderr:5]: Here too.

**Comment by Rageking8** [comment]
Haha 😂😂

**Comment by oli-obk** [comment]
@bors r+ rollup

Thanks both of you! :)

**Pull request merged**